### PR TITLE
tidy hooks

### PR DESCRIPTION
**Pull request by mapehe**
* Simplify some code in hooks.
* Get rid of some older UI errors in the console.
* Update swr to 2.0:
     * We get rid of custom `useMatchMutate` and use officially supported features instead.
     * We fix a bug I created earlier in #332.

### CODE REVIEW

File: lib/utils/matchMutate.ts [hunk 1]
```diff
@@ -1,23 +0,0 @@
-import { useSWRConfig } from "swr"
```
Comment (mapehe) [inline lib/utils/matchMutate.ts:1]: Using SWR 2.0 we can get rid of this custom logic.

File: components/common/Buttons.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import {
     IconButtonProps,
     StyleConfig,
 } from "@chakra-ui/react"
+import _ from "lodash"
```
Comment (LeadingMoominExpert) [inline components/common/Buttons.tsx:11]:
```suggestion
import { omit } from "lodash"
```

File: components/common/Buttons.tsx [hunk 3]
```diff
@@ -89,23 +90,14 @@ export const CustomButton = (props: ICustomButton): JSX.Element => {
...
-                props.hover
-                    ? {
-                          color: props.backgroundColor,
-                          backgroundColor: props.color,
-                      }
-                    : {}
-            }
-            {...props}
+            icon={<Icon as={props.icontype} boxSize="1.5rem" />}
+            {..._.omit(props, "icontype")}
```
Comment (LeadingMoominExpert) [inline components/common/Buttons.tsx:100]:
```suggestion
            {...omit(props, "icontype")}
```

File: lib/hooks/swrInterface.ts [hunk 4]
```diff
@@ -52,22 +57,21 @@ export const useGet = <T>(
     endpoint: string | null,
     deps: Endpoint[],
     user?: User,
-    params?: Record<string, string | number>
+    params?: Record<string, string | number | undefined>
 ): ApiGetResponse<T> => {
-    const key = getCacheKey(endpoint, deps, params)
+    const pars = _.pickBy(params, (x) => !_.isUndefined(x)) as Record<
```
Comment (LeadingMoominExpert) [inline lib/hooks/swrInterface.ts:62]:
```suggestion
    const pars = pickBy(params, (x) => !isUndefined(x)) as Record<
```

**Review by LeadingMoominExpert — APPROVED** [review]

**Pull request merged**